src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -177,11 +175,11 @@ public void Dispose()
 
         private (uint RefEscapeScope, uint ValEscapeScope) GetLocalScopes(LocalSymbol local)
         {
-            Debug.Assert(_localEscapeScopes?.ContainsKey(local) == true);
+            Debug.Assert(_localEscapeScopes?.ContainsKey(local) == true || _symbol != local.ContainingSymbol);
 
             return _localEscapeScopes?.TryGetValue(local, out var scopes) == true
                 ? scopes
-                : (CallingMethodScope, CallingMethodScope);
+                : (CurrentMethodScope, CallingMethodScope);
```

Comment:
BTW, does entering this branch usually mean we are in some kind of "use before definition" case?